src/main/java/com/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -44,27 +52,60 @@ public KeyVaultService(IEnvironmentReader environmentReader)
     logger.info("Auth type is " + this.authType);
 
     if (this.authType.equals(Constants.USE_MSI)) {
-      azureTokenCredentials = new MSICredentials(AzureEnvironment.AZURE);
+      credential = new DefaultAzureCredentialBuilder()
+          .excludeEnvironmentCredential()
+          .excludeAzureCliCredential()
+          .excludeSharedTokenCacheCredential()
+          .build();
     } else if (this.authType.equals(Constants.USE_CLI)) {
-      try {
-        azureTokenCredentials = AzureCliCredentials.create();
-      } catch (final IOException ex) {
-        logger.error(ex.getMessage());
-        throw ex;
-      }
+      credential = new DefaultAzureCredentialBuilder()
+        .excludeEnvironmentCredential()
+        .excludeManagedIdentityCredential()
+        .excludeSharedTokenCacheCredential()
+        .build();
     } else if (this.authType.equals(Constants.USE_MSI_APPSVC)) {
       try {
-        azureTokenCredentials = new AppServiceMSICredentials(AzureEnvironment.AZURE);
+        credential = new DefaultAzureCredentialBuilder()
+        .excludeEnvironmentCredential()
+        .excludeAzureCliCredential()
+        .excludeSharedTokenCacheCredential()
+        .build();
       } catch (final Exception ex) {
         logger.error(ex.getMessage());
         throw ex;
       }
     } else {
       this.authType = Constants.USE_MSI;
-      azureTokenCredentials = new MSICredentials(AzureEnvironment.AZURE);
+      credential = new DefaultAzureCredentialBuilder()
+      .excludeEnvironmentCredential()
+      .excludeAzureCliCredential()
+      .excludeSharedTokenCacheCredential()
+      .build();
     }
 
-    keyVaultClient = new KeyVaultClient(azureTokenCredentials);
+    //build secret clients
+    secretClient = new SecretClientBuilder()
+        .vaultUrl(getKeyVaultUri())
```

Comment:
rename secretClient to secretClientSync so we can differentiate. Otherwise, it is confusing.

Reply:
The name of the object from the SDK is directly reflected by the naming I used as they are named SecretClient and SecretAsyncClient.  My preference is to leave it as is and have parity with the SDK.

Reply:
I am not sure that having parity with the SDK is a good idea when we can actually improve code readability on our side of the fence as I believe it will be confusing to anyone new looking at the code. I know it confused me for a while before I figured out through the rest of the code that it was sync code.

Reply:
I dissent as I don't believe that forcing the reader to translate from one to another in order to know which SDK API is being called enhances clarity.

Reply:
I usually see Async appended. Can't remember seeing Sync used. @aflinchb - have you seen Sync used?

I do find it weird that it's not SecretClientAsync ... I almost always see at at the end.

Reply:
I literally named them the same as the SDK:

  private **SecretClient** secretClient;
  private **SecretAsyncClient** secretAsyncClient;